dev/run_session.py

```diff
@@ -1,65 +1,160 @@
 #!/usr/bin/env python3
 
-from os import environ 
+import csv
+import json
+from os import environ, path, remove
 
 from requests import post, Session
 
 
+UI_ADDRESS = "http://localhost:3002"
+API_ADDRESS = "http://localhost:3001"
+SIGN_UP_URL = f"{UI_ADDRESS}/auth/signup"
+SIGN_IN_URL = f"{UI_ADDRESS}/auth/signin"
+ASYNC_DL_URL = f"{API_ADDRESS}/api/cases/downloadAsync"
 CLIENT_EMAIL = environ.get("CLIENT_EMAIL", "fake@fake.fake")
 CLIENT_PASSWORD = environ.get("CLIENT_PASSWORD", "password")
 
-DL_PAYLOAD = {"format": "json", "query": "country:France"}
-
 SIGN_UP_PAYLOAD = {
     "email": CLIENT_EMAIL,
     "password": CLIENT_PASSWORD,
     "isAgreementChecked": True,
     "isNewsletterChecked": False
 }
+
 SIGN_IN_PAYLOAD = SIGN_UP_PAYLOAD.copy()
 SIGN_IN_PAYLOAD.update({
     "confirmEmail": CLIENT_EMAIL,
     "passwordConfirmation": CLIENT_PASSWORD 
 })
 
-UI_ADDRESS = "http://localhost:3002"
-API_ADDRESS = "http://localhost:3001"
-SIGN_UP_URL = f"{UI_ADDRESS}/auth/signup"
-SIGN_IN_URL = f"{UI_ADDRESS}/auth/signin"
-ASYNC_DL_URL = f"{API_ADDRESS}/api/cases/downloadAsync"
+# TODO: add special case for all.csv (redirected to another endpoint)
+DOWNLOADS = {
+    "france.json": {"format": "json", "query": "country:France"},
+    "france.csv": {"format": "csv", "query": "country:France"},
+    "france.tsv": {"format": "tsv", "query": "country:France"},
+    "all.json": {"format": "json"},
+    "all.tsv": {"format": "tsv"}
+}
+
+FIELDS_FILE = "../data-serving/scripts/export-data/functions/01-split/fields.txt"
 
 
-def sign_up(session):
+def sign_up(session: Session) -> None:
+    print(f"Signing up using email {CLIENT_EMAIL}")
     response = session.post(SIGN_UP_URL, json=SIGN_UP_PAYLOAD)
     print(response)
     print(response.text)
 
 
-def sign_in(session):
+def sign_in(session: Session) -> None:
+    print(f"Signing in using email {CLIENT_EMAIL}")
     response = session.post(SIGN_IN_URL, json=SIGN_IN_PAYLOAD)
     print(response)
     print(response.text)
 
 
-def download_cases(session, payload):
-    response = session.post(ASYNC_DL_URL, json=payload)
-    filename = f"{payload['format']}_test.{payload['format']}"
-    print(f"Saving to file {filename}")
-    with open(filename, 'wb') as fd:
-        for chunk in response.iter_content(chunk_size=128):
-            fd.write(chunk)
-
+def download_cases(session: Session) -> None:
+    print("Downloading cases")
+    for file_name, payload in DOWNLOADS.items():
+        print(f"Requesting cases for {payload}")
+        response = session.post(ASYNC_DL_URL, json=payload)
+        print(f"Saving to file {file_name}")
+        with open(file_name, "wb") as f:
+            for chunk in response.iter_content(chunk_size=128):
+                f.write(chunk)
+
+
+def load_case_fields() -> list[str]:
+    with open(FIELDS_FILE) as f:
+        contents = f.read()
+        return contents.splitlines()
+
+
+def check_files(fields: list[str]) -> None:
```

Comment:
list[str] syntax was introduced in Python 3.9, for 3.8 and below, it's List[str] with a from typing import List

Reply:
Yep, I run 3.9 locally and intend to put this and any other e2e scripts in a Docker container running 3.9.

Reply:
Ugh. I forgot we use 3.8 everywhere. Is this a vestige of using AWS Lambda, since that service does not support a python runtime more current than 3.8?
3.10 will release in October, and it would be good to stay current.

Reply:
Yes, Lambda doesn't support 3.9 https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html